radicle/src/profile.rs

```diff
@@ -26,87 +27,86 @@ pub enum Error {
     #[error(transparent)]
     Io(#[from] io::Error),
     #[error(transparent)]
-    KeyStore(#[from] crate::keystore::Error),
+    Keystore(#[from] crate::crypto::ssh::keystore::Error),
     #[error("no profile found at the filepath '{0}'")]
     NotFound(PathBuf),
-}
-
-#[derive(Debug)]
-pub struct UnsafeSigner {
-    pub public: PublicKey,
-    pub secret: SecretKey,
-}
-
-impl Signer for UnsafeSigner {
-    fn public_key(&self) -> &PublicKey {
-        &self.public
-    }
-
-    fn sign(&self, msg: &[u8]) -> Signature {
-        Signature(self.secret.sign(msg, None))
-    }
-
-    fn try_sign(&self, msg: &[u8]) -> Result<Signature, SignerError> {
-        Ok(self.sign(msg))
-    }
+    #[error("error connecting to ssh-agent: {0}")]
+    Agent(#[from] crate::crypto::ssh::agent::Error),
+    #[error("profile key `{0}` is not registered with ssh-agent")]
+    KeyNotRegistered(PublicKey),
 }
 
 #[derive(Debug)]
 pub struct Profile {
     pub home: PathBuf,
-    pub signer: UnsafeSigner,
     pub storage: Storage,
+    pub keystore: Keystore,
+    pub public_key: PublicKey,
 }
 
 impl Profile {
-    pub fn init(keypair: KeyPair) -> Result<Self, Error> {
+    pub fn init(passphrase: &str) -> Result<Self, Error> {
```

Comment:
Could be an Option to support unencrypted keys

Reply:
Yeah, this complicates a bunch of code paths so unless users ask for it, I think we should stick to encrypted keys.